### PR TITLE
cleanup vet invocation and vet errors

### PR DESCRIPTION
**Pull request by MHBauer**
- move vet outside of loop
 - use glide nv to determine directories to vet
 - cleanup vet errors in test directory

That loop rubs me the wrong way. This calls vet once vs in a loop.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -71,6 +71,8 @@ else
 	scBuildImageTarget = .scBuildImage
 endif
 
+NON_VENDOR_DIRS = $(shell $(DOCKER_CMD) glide nv)
```
Comment (MHBauer) [inline Makefile]: I forgot to check if this has to be defined after the `NO_DOCKER` `DOCKER_CMD` definition, but I figure it is clearer if it follows. Using `NO_DOCKER` still works.

File: Makefile [hunk 2]
```diff
@@ -209,9 +211,9 @@ verify: .init .generate_files verify-client-gen
 	    | grep -v v1alpha1/defaults.go); \
 	  do \
 	    golint --set_exit_status $$i; \
-	    go vet $$i; \
 	  done'
 	@#
+	$(DOCKER_CMD) go vet $(NON_VENDOR_DIRS)
```
Comment (duglin) [inline Makefile]: would using $NON_VENDOR_DIRS on the "golint" step work too? Does it result in checking the same set of files?

Comment (MHBauer) [inline Makefile]: It probably would. I'll check. `go vet` specifically was in my personal queue of 'annoyances'.

Comment (MHBauer) [inline Makefile]: The obvious thing here is that it won't be after the grep, so it'd check the things we want to ignore.

**Comment by duglin** [comment]
LGTM

File: Makefile (hunk 1 above)
Comment (MHBauer) [inline Makefile]: Thanks @arschles teaching me the `glide nv` trick.

**Pull request merged**